aws/lambda/pytorch-auto-revert/pytorch_auto_revert/autorevert_checker.py

```diff
@@ -311,25 +359,61 @@ def detect_autorevert_pattern_workflow(self, workflow_name: str) -> List[Dict]:
                 )
 
                 if not last_commit_with_same_job or not last_same_jobs:
-                    # No older commit with the same job found
+                    # No older commit with any jobs found
                     continue
 
-                if any(j.classification_rule == failure_rule for j in last_same_jobs):
-                    # The older commit has the same job failing with same rule
+                if any(
+                    j.classification_rule == failure_rule and j.conclusion == "failure"
+                    for j in last_same_jobs
+                ):
+                    # The older commit already has the same failure (regardless of job)
+                    continue
+
+                # Ensure there is some overlap in job coverage between suspected and older commit
+                older_coverage = list(
```

Comment:
this can only happen because you removed the check in https://github.com/pytorch/test-infra/pull/6983/files#diff-7174b1a731f38e3efb2765fac87a65ce7835d26a68cccd9c1e329e0b2070f1e2L285

If you keep the check, it is guaranteed that the `job_name` is present in both commits. This is due checks in `_fild_last_commit_with_job`

Reply:
the check is different (older commit vs newer commit), but your point is correct